### PR TITLE
fix #327 DataType has no field ninitialized

### LINKED ISSUE #327: Error `DataType has no field ninitialized` when saving Tuple
```julia
julia> import JLD2

julia> JLD2.save("tst.jld2", "x", (1,2))
Error encountered while save FileIO.File{FileIO.DataFormat{:JLD2}, String}("tst.jld2").

Fatal error:
ERROR: type DataType has no field ninitialized
...
```

 [033835bb] JLD2 v0.4.9

Julia: Version 1.7.0-DEV.1290, 6e8754337a*

EDIT:
The bug does *not* occur in
Julia 1.7.0-DEV.961, 2435f96f92*

### PR DESCRIPTION
**Pull request by jw3126**
fix #327

### CODE REVIEW

File: src/backwards_compatibility.jl [hunk 1]
```diff
@@ -35,3 +35,14 @@ end
         return isa(t, DataType) && t.mutable
     end
 end
+
+@static if :ninitialized in fieldnames(Tuple{})
```
Comment (jw3126) [inline src/backwards_compatibility.jl:39]: I don't think this needs to be `@static`, I just followed the `@static if VERSION < v"1.7.0-A"` code block above.

File: src/backwards_compatibility.jl [hunk 2]
```diff
@@ -35,3 +35,14 @@ end
...
+    # https://github.com/JuliaIO/JLD2.jl/issues/327
+    function ninitialized(@nospecialize(T::Type))::Int
+        T.ninitialized
+    end
+else
+    function ninitialized(@nospecialize(T::Type))::Int
```
Comment (jw3126) [inline src/backwards_compatibility.jl:45]: Not sure if there is a benefit of `@nospecialize` here, again I just followed the tradition `function ismutabletype(@nospecialize(t::Type))`

**Comment by JonasIsensee** [comment]
Hi @jw3126 ,
thanks a lot for taking care of this!

**Comment by jw3126** [comment]
@JonasIsensee thanks a lot for maintaining this essential package!

**Pull request merged**